### PR TITLE
chore: add workflow to rollout provider documentation

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Latest run: https://github.com/hashicorp/terraform-cdk/actions/runs/5688936110

### CODE REVIEW

File: .github/workflows/cdktf-provider-docs-rollout.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  schedule:
+    - cron: 0 0 * * 1
+
+jobs:
+  cdktfDocs:
+    strategy:
+      fail-fast: false
+      matrix:
+        provider:
+          # To onboard new providers, add a new entry to this matrix and make sure the team-tf-cdk github user has write permissions to the repository.
```
Comment (ansgarm) [inline .github/workflows/cdktf-provider-docs-rollout.yml:14]:
```suggestion
          # NOTE: This only works for Terraform providers in the HashiCorp Github org
          # To onboard new providers, add a new entry to this matrix and make sure the team-tf-cdk github user has write permissions to the repository.
```

Comment (ansgarm) [reply .github/workflows/cdktf-provider-docs-rollout.yml:14]: If this is true, I'd like to add this note (which will also serve future me 😄)

File: .github/workflows/cdktf-provider-docs-rollout.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+name: CDKTF Provider Docs Rollout
+
+on:
+  workflow_dispatch: {}
+  schedule:
+    - cron: 0 0 * * 1
```
Comment (ansgarm) [inline .github/workflows/cdktf-provider-docs-rollout.yml:6]: I'm bad at CRON and had to Google

```suggestion
    - cron: 0 0 * * 1 # Mondays at midnight
```

File: .github/workflows/cdktf-provider-docs-rollout.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          - repo: hashicorp/terraform-provider-null
+            fqn: "hashicorp/null"
+          - repo: hashicorp/terraform-provider-random
+            fqn: "hashicorp/random"
+          - repo: hashicorp/terraform-provider-time
+            fqn: "hashicorp/time"
+          - repo: hashicorp/terraform-provider-tls
+            fqn: "hashicorp/tls"
+
+    uses: hashicorp/terraform-cdk/.github/workflows/registry-docs-pr-based.yml@1b1be8ee471bb2845356e064e9f10b484f636cd9
```
Comment (ansgarm) [inline .github/workflows/cdktf-provider-docs-rollout.yml:36]: (optional) Do we need to pin if they point to the same repo? I could see future me trying to wonder why changes in the other workflow are not reflected 😅

**Review by ansgarm — APPROVED** [review]

File: .github/workflows/cdktf-provider-docs-rollout.yml (hunk 1 above)
Comment (DanielMSchmidt) [reply .github/workflows/cdktf-provider-docs-rollout.yml:14]: It’s not, if some other Provider repo gives push rights to our users it works for them as well

File: .github/workflows/cdktf-provider-docs-rollout.yml (hunk 3 above)
Comment (DanielMSchmidt) [reply .github/workflows/cdktf-provider-docs-rollout.yml:36]: No idea, I can try the local syntax and see if it works for security. But probably should

**Pull request merged**